docs/structured-streaming-kafka-integration.md

```diff
@@ -0,0 +1,185 @@
+---
+
+Structured Streaming integration for Kafka 0.10 to poll data from Kafka. It provides simple parallelism,
+1:1 correspondence between Kafka partitions and Spark partitions.
+
+### Linking
+For Scala/Java applications using SBT/Maven project definitions, link your application with the following artifact:
+
+    groupId = org.apache.spark
+    artifactId = spark-sql-kafka-0-10_{{site.SCALA_BINARY_VERSION}}
+    version = {{site.SPARK_VERSION_SHORT}}
+
+For Python applications, you need to add this above library and its dependencies when deploying your
+application. See the [Deploying](#deploying) subsection below.
+
+### Creating a Kafka Source Stream
+
+<div class="codetabs">
+<div data-lang="scala" markdown="1">
+
+    // Subscribe to 1 topic
+    spark
+      .readStream
+      .format("kafka")
+      .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
+      .option("subscribe", "topic1")
+      .load()
+
+    // Subscribe to multiple topics
+    spark
+      .readStream
+      .format("kafka")
+      .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
+      .option("subscribe", "topic1,topic2")
+      .load()
+
+    // Subscribe to a pattern
+    spark
+      .readStream
+      .format("kafka")
+      .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
+      .option("subscribePattern", "topic.*")
+      .load()
+
+</div>
+<div data-lang="java" markdown="1">
+
+    // Subscribe to 1 topic
+    spark
+      .readStream()
+      .format("kafka")
+      .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
+      .option("subscribe", "topic1")
+      .load()
+
+    // Subscribe to multiple topics
+    spark
+      .readStream()
+      .format("kafka")
+      .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
+      .option("subscribe", "topic1,topic2")
+      .load()
+
+    // Subscribe to a pattern
+    spark
+      .readStream()
+      .format("kafka")
+      .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
+      .option("subscribePattern", "topic.*")
+      .load()
+
+</div>
+<div data-lang="python" markdown="1">
+
+    # Subscribe to 1 topic
+    spark
+      .readStream()
+      .format("kafka")
+      .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
+      .option("subscribe", "topic1")
+      .load()
+
+    # Subscribe to multiple topics
+    spark
+      .readStream
+      .format("kafka")
+      .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
+      .option("subscribe", "topic1,topic2")
+      .load()
+
+    # Subscribe to a pattern
+    spark
+      .readStream()
+      .format("kafka")
+      .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
+      .option("subscribePattern", "topic.*")
+      .load()
+
+</div>
+</div>
+
+Each row in the source has the following schema:
+<table class="table">
+<tr><th>Column</th><th>Type</th></tr>
+<tr>
+  <td>key</td>
+  <td>binary</td>
+</tr>
+<tr>
+  <td>value</td>
+  <td>binary</td>
```

Comment:
This should probably include an example of how to select key / value in the desired format.